reactor-netty-core/src/main/java/reactor/netty/resources/PooledConnectionProvider.java

```diff
@@ -334,6 +335,10 @@ protected PoolFactory<T> poolFactory(SocketAddress remoteAddress) {
 		return poolFactoryPerRemoteHost.getOrDefault(remoteAddress, defaultPoolFactory);
 	}
 
+	protected ConnectionPoolMetrics delegateConnectionPoolMetrics(InstrumentedPool.PoolMetrics metrics) {
+		return new DelegatingConnectionPoolMetrics(metrics);
+	}
```

Comment:
Separate the method in `Http2ConnectionProvider` and `Http3ConnectionProvider`, which are implementations of `PooledConnectionProvider`, to enable the creation of `DelegatingConnectionPoolMetrics` implementations.